projects/packages/stats-admin/src/class-rest-controller.php

```diff
@@ -41,33 +58,41 @@ public function register_rest_routes() {
 				'permission_callback' => array( $this, 'check_user_privileges_callback' ),
 			)
 		);
+
+		// Stats for a resource type.
 		register_rest_route(
 			static::$namespace,
 			sprintf( '/sites/%d/stats/(?P<resource>[\-\w]+)', Jetpack_Options::get_option( 'id' ) ),
 			array(
 				'methods'             => WP_REST_Server::READABLE,
-				'callback'            => array( $this, 'get_stats_resource_from_wpcom' ),
+				'callback'            => array( $this, 'get_stats_resource' ),
 				'permission_callback' => array( $this, 'check_user_privileges_callback' ),
 			)
 		);
+
+		// Single post info.
 		register_rest_route(
 			static::$namespace,
-			sprintf( '/sites/%d/(?P<resource>[\-\w]+)/(?P<resource_id>[\d]+)/(?P<sub_resource>[\-\w]+)', Jetpack_Options::get_option( 'id' ) ),
+			sprintf( '/sites/%d/posts/(?P<resource_id>[\d]+)', Jetpack_Options::get_option( 'id' ) ),
 			array(
 				'methods'             => WP_REST_Server::READABLE,
-				'callback'            => array( $this, 'get_site_sub_resource_from_wpcom' ),
+				'callback'            => array( $this, 'get_single_post' ),
 				'permission_callback' => array( $this, 'check_user_privileges_callback' ),
 			)
 		);
+
+		// Single post likes.
 		register_rest_route(
 			static::$namespace,
-			sprintf( '/sites/%d/(?P<resource>[\-\w]+)/(?P<resource_id>[\d]+)', Jetpack_Options::get_option( 'id' ) ),
+			sprintf( '/sites/%d/posts/(?P<resource_id>[\d]+)/likes', Jetpack_Options::get_option( 'id' ) ),
 			array(
 				'methods'             => WP_REST_Server::READABLE,
-				'callback'            => array( $this, 'get_site_single_resource_from_wpcom' ),
+				'callback'            => array( $this, 'get_single_post_likes' ),
 				'permission_callback' => array( $this, 'check_user_privileges_callback' ),
 			)
 		);
+
+		// Required site endpoints for the Stats app.
```

Comment:
It feels we could completely remove this endpoint, right?
Possible resources are:
`site-has-never-published-post` -> Why not call the corresponding WPCOM v2 endpoint for this case? 

Then:
```
'posts':
'sharing-buttons':
'plugins':
'keyrings':
'rewind':
```
which you plan to remove from the frontend, so not needed.

And finally `stats`, which ideally should be handled the same with the rest stats endpoints

Reply:
> I get your reasoning but in this case I believe it's best to keep a single source of truth, therefore I'd suggest using the WPCOM v2 endpoint.

Addressed in a170723e03.

> Now about the endpoints that you plan to remove on the next stage, what's the harm in returning a 404? Is there a specific reason to register those just to return an empty response?

There were certain endpoints that stops the rendering of the frontend if not stubbed, and I also tried to make the console less noisy, but it seems the frontend works okay with all the stubs removed. Addressed in a170723e03

> I'd also recommend to explicitly register only the ones you need along with the corresponding permissions.

Addressed in a170723e03.

> Note that although site-has-never-published-post only needs read permissions normally, under the context of accessing site stats it's fine to use check_user_privileges_callback.

Yes. However the WPCOM endpoint requires user permission and I spun up a diff `D93971-code`. Could you please take a look at that as well? Thanks.

>  HOWEVER, keep in mind that if you plan to add more endpoints in the future, in case an endpoint needs admin permissions, you have to explicitly register it and make sure to add a new permission_callback,

Yeah for sure. Thanks for the reminder.